src/fields/field_variable.js

```diff
@@ -139,16 +139,23 @@ class FieldVariable extends Blockly.FieldVariable {
   showEditor_(event) {
     super.showEditor_(event);
     const sourceBlock = this.getSourceBlock();
+    const style = sourceBlock.style;
     if (sourceBlock.isShadow()) {
-      sourceBlock.setColour(sourceBlock.style.colourQuaternary);
+      this.originalStyle = sourceBlock.getStyleName();
```

Comment:
For all three fields: `originalStyle` should be declared as a property on the class if it hasn't been already (instead of just adding it to the object because js will let you).

Reply:
Done